GameServerLib/GameObjects/Stats/Stats.cs

```diff
@@ -217,11 +218,23 @@ public void LevelUp()
             CurrentHealth = HealthPoints.Total / (HealthPoints.Total - HealthPerLevel) * CurrentHealth;
             ManaPoints.BaseValue = ManaPoints.Total + ManaPerLevel;
             CurrentMana = ManaPoints.Total / (ManaPoints.Total - ManaPerLevel) * CurrentMana;
-            AttackDamage.BaseValue = AttackDamage.BaseValue + AdPerLevel;
-            Armor.BaseValue = Armor.BaseValue + ArmorPerLevel;
-            MagicResist.BaseValue = MagicResist.Total + MagicResistPerLevel;
-            HealthRegeneration.BaseValue = HealthRegeneration.BaseValue + HealthRegenerationPerLevel;
-            ManaRegeneration.BaseValue = ManaRegeneration.BaseValue + ManaRegenerationPerLevel;
+            AttackDamage.BaseValue += AdPerLevel;
+            Armor.BaseValue += ArmorPerLevel;
+            MagicResist.BaseValue += MagicResistPerLevel;
+            HealthRegeneration.BaseValue += HealthRegenerationPerLevel;
+            ManaRegeneration.BaseValue += ManaRegenerationPerLevel;
+
+            if (Level > 1)
+            {
+                AttackSpeedMultiplier = new Stat
+                (
+                    AttackSpeedMultiplier.BaseValue,
```

Comment:
Why not using StatModifier with only PercentBaseBonus? It will be clearer and more modular.